ENIGMAsystem/SHELL/Platforms/xlib/LINUXjoystick.cpp

```diff
@@ -197,10 +206,13 @@ namespace enigma_user
   }
 
   bool joystick_exists(int id) {
-    if (size_t(id) >= enigma::joysticks.size())
-      return false;
-    const enigma::e_joystick * const js = enigma::joysticks[id].get();
-    return js && js->device != -1;
+    checkPositiveId(false);
+    if (size_t(id) < enigma::joysticks.size()) {
+      const enigma::e_joystick * const js = enigma::joysticks[id].get();
+      if (js && js->device != -1) return true;
+    }
+    struct stat ignore;
+    return (stat(device_name(id).c_str(), &ignore) == 0);
```

Comment:
I will say this is a little bothersome because it breaks from the resource adding tradition. For example, `sprite_exists` only tells me whether I've added (e.g, loaded) the sprite, it does not tell me if the sprite exists on disk. This could perhaps be broken off into `joystick_is_loaded` or something.